client/app/bundles/course/video/submission/actions/video.js

```diff
@@ -139,3 +140,40 @@ export function seekStart() {
 export function seekEnd() {
   return { type: videoActionTypes.SEEK_END };
 }
+
+/**
+ * Creates an action to remove events from the state store.
+ * @param sequenceNums The event sequence numbers to remove
+ * @return {{type: videoActionTypes, sequenceNums: Set<number>}}
+ */
+function removeEvents(sequenceNums) {
+  return { type: sessionActionTypes.REMOVE_EVENTS, sequenceNums };
+}
+
+/**
+ * Sends current events to the server.
+ *
+ * Events will be removed from the state store if the API call is successful.
+ * @return {function(*, *)}
+ */
+export function sendEvents() {
+  return (dispatch, getState) => {
+    const videoState = getState().video;
+    const sessionId = videoState.sessionId;
+
+    if (sessionId === null) {
+      return;
```

Comment:
Just to check, this is just precautionary? By right if sessionId is null there shouldn't be events being generated and pushed onto the queue?

Reply:
Yup, but I send a request to the server even if there are no events, so as to update the `session_end` time. If `sessionId` is null, we skip everything to not spam the server with unnecessary requests.